### PR TITLE
fix: Updating the logic to render 'allUsers' having access to the workspace

### LINKED ISSUE #28013: [Bug]: When new member is added to a workspace, front-end does not render updated user list
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Description

When a new member is successfully added to a workspace, the front end does not render the updated user list on /applications page. But the same works on the dedicated manage users page. 

For applications page:
[POST] /invite is called
[GET] /member is called after addition
But it is not put back in the state or a rerender is not done even after the modal is closed.

image

### Steps To Reproduce

1. Open /applications page.
2. Create a workspace if none is available.
3. Share the workspace with another email.
4. Close the Share modal.

### Public Sample App

_No response_

### Environment

Production

### Issue video log

_No response_

### Version

Cloud

### PR DESCRIPTION
**Pull request by yasharma2301**
#### PR fixes following issue(s)
Fixes #28013 and ensures that it does not break #17728

#### Type of change
- Bug fix (non-breaking change which fixes an issue)

#### How Has This Been Tested?
- [x] Manual
- [ ] JUnit
- [ ] Jest
- [ ] Cypress

>

#### Dev activity
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [ ] PR is being merged under a feature flag

#### QA activity:
- [ ] [Speedbreak features](https://github.com/appsmithorg/TestSmith/wiki/Guidelines-for-test-plans#speedbreakers-) have been covered
- [ ] Test plan covers all impacted features and [areas of interest](https://github.com/appsmithorg/TestSmith/wiki/Guidelines-for-test-plans#areas-of-interest-)
- [ ] Test plan has been peer reviewed by project stakeholders and other QA members
- [ ] Manually tested functionality on DP
- [ ] We had an implementation alignment call with stakeholders post QA Round 2
- [ ] Cypress test cases have been added and approved by SDET/manual QA
- [ ] Added `Test Plan Approved` label after Cypress tests were reviewed
- [ ] Added `Test Plan Approved` label after JUnit tests were reviewed

### CODE REVIEW

File: app/client/src/pages/common/SharedUserList.tsx [hunk 1]
```diff
@@ -45,9 +46,11 @@ export default function SharedUserList(props: any) {
     });
 
     const { users } = workspace;
-    const newUsers = users?.filter((user: any) => user.userId) || [];
-    return convertUsersToAvatar(newUsers);
+    // The uniqBy filter handles the case where same user is added multiple times
+    const uniqueUsers = uniqBy(users || [], "username");
```
Comment (ankitakinger) [inline app/client/src/pages/common/SharedUserList.tsx:50]: We are going to handle it on the backend, so we can avoid using the lodash function here. There is no filtering required here.

Comment (yasharma2301) [reply app/client/src/pages/common/SharedUserList.tsx]: Cool, updated the PR accordingly

**Review by ankitakinger — APPROVED** [review]
LGTM

**Pull request merged**